demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,1022 @@
+
+/**
+ * @brief Send an HTTP request based on a specified method and path, and
+ * print the response received from the server.
+ *
+ * @param[in] pHost The host name of the server.
+ * @param[in] hostLen The length of pHost.
+ * @param[in] pRequest The HTTP Request-URI.
+ * @param[in] requestLen The length of pRequest.
+ * @param[in] fileSize The length of the file at S3_PRESIGNED_GET_URL.
+ * @param[in] requestQueue The queue to which HTTP requests should be written.
+ * @param[in] responseQueue The queue from which HTTP responses should be read.
+ *
+ * @return false on failure; true on success.
+ */
+static bool downloadS3ObjectFile( const char * pHost,
+                                  const size_t hostLen,
+                                  const char * pRequest,
+                                  const size_t requestLen,
+                                  mqd_t requestQueue,
+                                  mqd_t responseQueue );
+
+/**
+ * @brief Enqueue an HTTP request for a range of the S3 file.
+ *
+ * @param[in] requestInfo The #HTTPRequestInfo_t for configuring the request.
+ * @param[in] requestQueue The queue to which HTTP requests should be written.
+ * @param[in] start The position of the first byte in the range.
+ * @param[in] end The position of the last byte in the range, inclusive.
+ *
+ * @return QUEUE_OP_FAILURE on failure; QUEUE_OP_WOULD_BLOCK if would block,
+ * QUEUE_OP_SUCCESS on success.
+ */
+static QueueOpStatus_t requestS3ObjectRange( const HTTPRequestInfo_t * requestInfo,
+                                             mqd_t requestQueue,
+                                             const size_t start,
+                                             const size_t end );
+
+
+/**
+ * @brief Processes an HTTP response from the response queue.
+ *
+ * @param[in] responseQueue The queue from which HTTP responses should be read.
+ * @param[out] responseItem The HTTP response recieved.
+ *
+ * @return QUEUE_OP_FAILURE on failure; QUEUE_OP_WOULD_BLOCK if would block,
+ * QUEUE_OP_SUCCESS on success.
+ */
+static QueueOpStatus_t retrieveHTTPResponse( mqd_t responseQueue,
+                                             ResponseItem_t * responseItem );
+
+/**
+ * @brief Retrieve the size of the S3 object that is specified in pPath.
+ *
+ * @param[in] requestInfo The #HTTPRequestInfo_t for configuring the request.
+ * @param[in] requestQueue The queue to which HTTP requests should be written.
+ * @param[in] responseQueue The queue from which HTTP responses should be read.
+ * @param[out] pFileSize - The size of the S3 object.
+ *
+ * @return false on failure; true on success.
+ */
+static bool getS3ObjectFileSize( const HTTPRequestInfo_t * requestInfo,
+                                 mqd_t requestQueue,
+                                 mqd_t responseQueue,
+                                 size_t * pFileSize );
+
+/**
+ * @brief Enqueue HTTP requests based on a specified method and path.
```

Comment:
Is this comment correct? I think this task routine dequeues HTTP requests and enqueues HTTP responses.